let Joi,bcrypt,User,CustomErrorHandler;_ffd‍.x([["default",()=>_ffd‍.o]]);_ffd‍.w("joi",[["default",["Joi"],function(v){Joi=v}]]);_ffd‍.w("bcrypt",[["default",["bcrypt"],function(v){bcrypt=v}]]);_ffd‍.w("../../models",[["User",["User"],function(v){User=v}]]);_ffd‍.w("../../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);




const signupController = {
    async signup(req,res,next){

        // validation
        
        const signupSchema = Joi.object({
            email:Joi.string().email().required(),
            phone:Joi.string().min(10).required(),
            password:Joi.string().regex(/^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/).required()
        })

        const {error} = signupSchema.validate(req.body);
        if (error) {
            return next(error)
        }

        //check if user is already exits in database

        try {
            const exit = await User.exists({email:req.body.email});
            if (exit) {
                return next(CustomErrorHandler.alredayExits('email is already exit'))
            }
        } catch (err) {
            return next(err);
        }

        const {email,phone,password} = req.body;

        //hashed the password

        const hashedPassword = await bcrypt.hash(password,10);

        //prepare the model

        const user = new User({
            email,
            phone,
            password:hashedPassword
        })

        try {
            const result = await user.save();
        } catch (err) {
            return next(err)
        }

        res.json({status:1,message:'sucessfully signup'})
    }
}

_ffd‍.d(signupController);